packages/cli-kit/src/public/node/git.ts

```diff
@@ -83,8 +84,14 @@ export function addToGitIgnore(root: string, entry: string): void {
   const gitIgnoreContent = readFileSync(gitIgnorePath).toString()
   const eol = detectEOL(gitIgnoreContent)
 
-  if (gitIgnoreContent.split(eol).some((line) => line.trim() === entry.trim())) {
-    // The file already existing in the .gitignore
+  const lines = gitIgnoreContent.split(eol).map((line) => line.trim())
+  const ignoreManager = ignore.default({allowRelativePaths: true}).add(lines)
+
+  const isIgnoredEntry = ignoreManager.ignores(joinPath(entry))
+  const isIgnoredEntryAsDir = ignoreManager.ignores(joinPath(entry, 'ignored.txt'))
```

Comment:
What's `ignored.txt`?

Reply:
It's a mocked file entry that supports the following two test scenarios:
```
- does nothing when .shopify/* pattern exists in .gitignore
- does nothing when .shopify/** pattern exists in .gitignore
```

The ignore package doesn't interact with the file system; it just analyzes paths and patterns. This mocked entry helps us ensure that `.shopify/*` and `.shopify/**` are really working as expected.